pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,373 @@
+        function hide_progress_message() {
+            is_deploying.hide();
+        }
+
+        function remove_notifications() {
+            $('div.container-fluid.alert').remove();
+            deploy_dialog_remove_errors();
+        }
+
+
+        function action_in_progress() {
+            return (is_deploying.is(':visible'));
+        }
+
+        return {
+            'deploy_app': deploy_app
+        };
+
+    }
+
+    function deploy_app() {
+        //alert("deploy_app")
+        var jsondata = "";
+        deploy_dialog_remove_errors();
+        jsondata = appdeployer.jsondata;
+
+        var ns = $('#deploy-app-namespace-field').val();
+        if ($('#deploy-app-namespace-field').val() === 'Custom Namespace')
+            ns = $('#deploy-app-namespace-field-custom').val().trim();
+
+        var has_errors = false;
+        if (jsondata === '') {
+            $('#deploy-app-manifest-file-empty').show();
+            $('#deploy-app-manifest-file').parent().addClass('has-error');
+            has_errors = true;
+        }
+        if (ns.trim() === '' || ns.trim() === 'Enter Namespace Here') {
+            $('#deploy-app-namespace-field-note').show();
+            $('#deploy-app-namespace-field-custom').parent().addClass('has-error');
+            has_errors = true;
+        }
+        if (!has_errors)
+            appdeployer.manager.deploy_app(ns, jsondata);
+    }
+
+    function deploy_dialog_remove_errors() {
+        $('#deploy-app-manifest-file-note').hide();
+        $('#deploy-app-manifest-file-note-details').hide();
+        $('#deploy-app-namespace-field-note').hide();
+        $('#deploy-app-namespace-field-note-details').hide();
+        $('#deploy-app-general-error').hide();
+        $('#deploy-app-manifest-file-empty').hide();
+        $('#deploy-app-namespace-field-custom-empty').hide();
+        $('#deploy-app-namespace-field-custom').parent().removeClass('has-error');
+        $('#deploy-app-manifest-file').parent().removeClass('has-error');
+
+    }
+
+    function pre_init() {
+        //alert("pre_init")
+        var firstTime = true;
+        var dlg = $('#deploy-app-dialog');
+        var btn = $('#deploy-app-start');
+        var manifest_file = $('#deploy-app-manifest-file');
+        var manifest_file_note = $('#deploy-app-manifest-file-note');
+        var manifest_file_details = $("#deploy-app-manifest-file-note-details");
+        var ns_selector = $('#deploy-app-namespace-field');
+        appdeployer.jsondata = "";
+        var text = "";
+
+        btn.on('click', function() {
+            deploy_app();
+        });
+
+        dlg.on('show.bs.modal', function() {
+            //alert("show.bs.models");
+            if (firstTime) {
+                var optionls = [];
+                var nslist = get_kentities("namespaces");
+                for (var i = 0; i < nslist.length; i++) {
+                    optionls.push('<option translatable="yes" value="' + nslist[i] + '">' + nslist[i] + '</option>');
```

Comment:
Cross site scripting attack. We shouldn't assemble HTML like this with values received from the server.